### PR TITLE
Feature rockets. Put in some rocket engines!  Adding in the bell x1 vehicle (not mission).  Added ability to add internal engine that adds no drag.

### PR DESCRIPTION
**Pull request by WallyMaier**
# Pull Request Checklist
 * Merge in the latest Develop branch changes to your branch
 * Remove .pyc files from your repository:
   * Linux/Mac
     * find . -name *.pyc -delete
   * Windows
     * del /S *.pyc
 * Run automatic regression and makes sure everything is passing
 * Did you make a new regression test that covers your new code?
 * Did you update your docstrings?
 * Did you update your headers to include your name and date?
 * Do a final compare with the Develop branch to be sure what you're changing

### CODE REVIEW

**Comment by WallyMaier** [comment]
pretty sure @planes last pull request breaks the weight regression

**Comment by WallyMaier** [comment]
avl passes on my machine.....not sure what is going on...

**Comment by WallyMaier** [comment]
I give up

File: regression/scripts/rocket_network/Rocketdyne_F1.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    vac                   = conditions.freestream
+    vac.altitude          = ones_1col*0.0
+    
+    atmosphere            = SUAVE.Analyses.Atmospheric.US_Standard_1976()
+    atmo_data             = atmosphere.compute_values(vac.altitude,0,True) 
+    
+    vac.gravity           = ones_1col*9.81
+    vac.pressure          = ones_1col*0.0
+    vac.temperature       = ones_1col*atmo_data.temperature
+    vac.density           = ones_1col*atmo_data.density
```
Comment (planes) [inline regression/scripts/rocket_network/Rocketdyne_F1.py]: with this density isn't zero? Does that matter?

File: trunk/SUAVE/Attributes/Planets/Earth.py [hunk 2]
```diff
@@ -42,9 +42,9 @@ def __defaults__(self):
         None
         """         
         self.tag = 'Earth'
-        self.mass              = 5.98e24  # kg
-        self.mean_radius       = 6.371e6  # m
-        self.sea_level_gravity = 9.80665  # m/s^2   
+        self.mass              = 5.98e24  # [kg]
```
Comment (planes) [inline trunk/SUAVE/Attributes/Planets/Earth.py]: don't just make comments, actually change the code! At least put in Units in

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Throttle_Constant_Speed.py [hunk 3]
```diff
@@ -73,9 +73,9 @@ def __defaults__(self):
         
         # initials and unknowns
         ones_row = self.state.ones_row
-        self.state.unknowns.body_angle = ones_row(1) * 5.0 * Units.deg
-        self.state.unknowns.wind_angle = ones_row(1) * 0.0 * Units.deg
-        self.state.residuals.forces    = ones_row(2) * 0.0
+        self.state.unknowns.body_angle      = ones_row(1) * 5.0 * Units.deg
```
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Throttle_Constant_Speed.py]: Revert this file. Unnecessary change

File: trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/compressibility_drag_total.py [hunk 4]
```diff
@@ -86,8 +86,8 @@ def compressibility_drag_total(state,settings,geometry):
 
         # Get main fuselage data - note that name of fuselage is important here
         # This should be changed to be general 
-        main_fuselage = fuselages['fuselage']
-
+        #main_fuselage = fuselages['fuselage']
```
Comment (planes) [inline trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/compressibility_drag_total.py]: delete the commented out code and move the comments up

File: regression/scripts/AVL/test_AVL.py [hunk 5]
```diff
@@ -93,7 +93,7 @@ def main():
     
     # moment coefficient check
     moment_coefficient            = results.conditions.cruise.stability.static.CM[0][0]
-    moment_coefficient_true       = -0.620326644
+    moment_coefficient_true       = -0.620903460878
```
[the remark this answers is not in the document]
Comment (WallyMaier) [reply regression/scripts/AVL/test_AVL.py]: these were grabbed from your recent pull request....the Data ordering

File: regression/scripts/Vehicles/Bell_X1.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+    # mass properties
+    vehicle.mass_properties.max_takeoff               = 12250.0 * Units.pounds 
+    vehicle.mass_properties.takeoff                   = 12250.0 * Units.pounds   
+    vehicle.mass_properties.operating_empty           = 7150.0  * Units.pounds 
+    vehicle.mass_properties.max_zero_fuel             = 7150.0  * Units.pounds 
+    vehicle.mass_properties.cargo                     = 150.0   * Units.pounds  
+
+    # envelope properties
+    vehicle.envelope.ultimate_load = 2.5
```
[the remark this answers is not in the document]
Comment (WallyMaier) [reply regression/scripts/Vehicles/Bell_X1.py]: which ones?

File: regression/scripts/Vehicles/Boeing_737.py [hunk 7]
```diff
@@ -126,27 +126,22 @@ def vehicle_setup():
...
-    
-    wing.spans.projected         = 14.2      #
-
+    wing.spans.projected         = 14.2      
     wing.chords.root             = 4.7
     wing.chords.tip              = .955   
     wing.chords.mean_aerodynamic = 8.0
-
-    wing.areas.reference         = 32.488    #
-    wing.areas.exposed           = 199.7792                  # Exposed area of the horizontal tail
```
Comment (WallyMaier) [inline regression/scripts/Vehicles/Boeing_737.py]: These cannot be deleted

File: regression/scripts/rocket_network/Rocketdyne_J2.py [hunk 8]
```diff
@@ -0,0 +1,205 @@
+# Rocketdyne_J2.py
```
Comment (planes) [inline regression/scripts/rocket_network/Rocketdyne_J2.py]: Wait do these regressions cover different things?

At first glance they look similar.

File: trunk/SUAVE/Components/Energy/Converters/Rocket_Combustor.py [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        thr    = conditions.propulsion.throttle 
+        
+                
+        # unpacking values from self
+        gamma = self.propellant_data.isentropic_expansion_factor
+        Tc    = self.propellant_data.combustion_temperature
+        R     = self.propellant_data.gas_specific_constant
+        
+        #--Compute combustion quantities--
+        # This is where the combustion surrogate would be utilized
```
Comment (planes) [inline trunk/SUAVE/Components/Energy/Converters/Rocket_Combustor.py]: Is this incomplete and not used?

File: trunk/SUAVE/Components/Energy/Converters/de_Laval_Nozzle.py [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        expansion_ratio = self.expansion_ratio
+        area_throat     = self.area_throat 
+        
+        
+        #--Method for computing the nozzle properties--
+        
+        #--Converging Nozzle --
+        # getting stagnation quantities at throat
+        Pt_throat   = Pt_combustion*pi
+        Tt_throat   = Tt_combustion*pi**((gamma-1.)/(gamma)*etapold)
```
Comment (planes) [inline trunk/SUAVE/Components/Energy/Converters/de_Laval_Nozzle.py]: if you expect to run this calculation often you could pre-calculate gamma-1 as gamma_m_ and... so on. But it's not a huge necessity

File: trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+        Ae            = area_throat*expansion_ratio
+        mdot_temp     = throttle*rho*exhaust_velocity*Ae
+        
+        #--Computing Dimensional Thrust
+        thrust        = num_eng*mdot_temp*C
+        
+        #--Make mdot size of thrust
+        mdot               = mdot_temp 
+        mdot[mdot<0.0]     = 0.0 
+        thrust[thrust<0.0] = 0.0
```
Comment (planes) [inline trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py]: lol, come on no negative thrust!...

actually, you may find removing this could help convergence.

File: regression/scripts/rocket_network/Rocketdyne_J2.py (hunk 8 above)
Comment (WallyMaier) [reply regression/scripts/rocket_network/Rocketdyne_J2.py]: They both regress the liquid rocket network, but with different propellant types. Im not sure if a propellant regression should be created (seems kinda dumb) or another rocket (overkill).  Ill go with what you think is best.

File: trunk/SUAVE/Components/Energy/Converters/Rocket_Combustor.py (hunk 9 above)
Comment (WallyMaier) [reply trunk/SUAVE/Components/Energy/Converters/Rocket_Combustor.py]: Currently it is placeholder for more advanced propulsion data to be linked.  So it a subtle reminded of worked needed to be done.  Should I remove and make a github project/issue?

File: trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py (hunk 11 above)
Comment (WallyMaier) [reply trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py]: Should it be removed as it stands currently?   I will try comment these in the X1 case (im not hopeful) hahah.

File: trunk/SUAVE/Methods/Missions/Segments/Common/Weights.py [hunk 12]
```diff
@@ -71,14 +71,18 @@ def update_gravity(segment,state):
...
+    alt    = state.conditions.freestream.altitude
+    
+    # Calculate
+    Rh     = alt+R
+    g      = g0*(R/(Rh*Rh))  # [m/s^2] 
 
-    # pack
-    state.conditions.freestream.gravity[:,0] = g
+    # Pack
+    state.conditions.freestream.gravity[:,0] = g0
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Common/Weights.py]: Is the newly calculated g used here?

File: regression/scripts/rocket_network/Rocketdyne_F1.py [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+    ones_1col             = np.ones([1,1])   
+    conditions.freestream     = Data()
+    conditions.propulsion = Data()
+    
+    # vacuum conditions
+    vac                   = conditions.freestream
+    vac.altitude          = ones_1col*0.0
+    
+    atmosphere            = SUAVE.Analyses.Atmospheric.US_Standard_1976()
+    atmo_data             = atmosphere.compute_values(vac.altitude,0,True)
```
Comment (timdmacdo) [inline regression/scripts/rocket_network/Rocketdyne_F1.py]: Why are you calling the atmosphere model for the vacuum case?

File: trunk/SUAVE/Methods/Missions/Segments/Common/Weights.py (hunk 12 above)
Comment (WallyMaier) [reply trunk/SUAVE/Methods/Missions/Segments/Common/Weights.py]: No, there was a reason, but i cant remember what it was.  If you cant think of  a reason, ill update

File: trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py (hunk 11 above)
Comment (WallyMaier) [reply trunk/SUAVE/Components/Energy/Processes/Rocket_Thrust.py]: doesn't help x1 convergence
